src/pypromice/tx/get_watsontx.py

```diff
@@ -98,13 +98,16 @@ def get_watsontx():
             name=None
             d=None
         
-        if name and 'Watson station' in name:
-            print(f'Watson station message, {d.strftime("%Y-%m-%d %H:%M:%S")}')
+        if name and 'Watson' in name or name and 'GIOS' in name:
+            print(f'Watson/GIOS station message, {d.strftime("%Y-%m-%d %H:%M:%S")}')
+
             l0 = L0tx(message, formatter_file, type_file, 
                       sender_name=['emailrelay@konectgds.com','sbdservice'])
             
             if l0.msg: 
-                out_fn = 'watson_station_tx.txt'
+                content, attachment = l0.getEmailBody()
+                attachment_name = str(attachment.get_filename())
+                out_fn = attachment_name.split('able')[0]+'able.txt'
```

Comment:
What is the expected filename of the attached file? Does it already have an extension? Is the substring `able` always present in the file name?

If attachment_name contains an extension and 'able' is not present, then will out_fn include the original extension before the suffix `able.txt`.

Example:
```
>> attachment_name = "the_original_filename.txt" 
>> out_fn = attachment_name.split('able')[0]+'able.txt'
>> out_fn
"the_original_filename.txtable.txt"
```

Consider adding some comments if this is an unexpected case.

Reply:
There are currently four stations that transmit. These are the message subjects in the transmission and the filename of the attachment:

|      Message subject          |     Attachment filename                      |
|------------------------------------------|:-----------------------------------------------:|
|      "Data from Watson station"  |    Watson10MinTable<number>.dat       |
|      "Data from GIOS station"    |    GIOS_Russel_table<number>.dat        |
|      "Data from GIOS station"    |    GIOS_Watson_L_table<number>.dat  |
|      "Data from GIOS station"    |    GIOS_LR_table<number>.dat                |

Where `<number>` refers to the file number, which increases sequentially. So I want to make a txt file that all data from one station can be compiled into, e.g. `GIOS_Russel_table001.dat, GIOS_Russel_table002.dat, GIOS_Russel_table003.dat >> GIOS_Russel_table.txt`.

It's a very clunky solution, but I split the .dat filename at "able" like this:

```
>> attachment_name = "GIOS_LR_table001.dat" 
>> out_fn = attachment_name.split('able')[0]+'able.txt'
>> out_fn
"GIOS_LR_table.txt"
```

It works for the current set-up but will need changing in the future if the naming convention of the transmission attachment files changes. Maybe we could implement something that removes the message number and replaces the suffix `.dat` to `.txt` with `pathlib`?

Reply:
With pathlib you can use the property `stem` to get the name without extension or the method `with_suffix`
```python
out_fn: str = Path(attachment_name).stem + '.txt'
```
or
```python
out_fn: Path = Path(attachment_name).with_suffix('.txt')
```
With `os.path` you can use the function `splitext`:
```python
out_fn: str = os.path.splitext(attachment_name)[0] + '.txt'
```

Reply:
I would prefer one of those solutions in favour of the "able" split because it is more clear what is happening.

Reply:
New solution @ladsmund:
https://github.com/GEUS-Glaciology-and-Climate/pypromice/blob/8d244f185212d0b8609047800f12128c0f82bbd2/src/pypromice/tx/get_watsontx.py#L110-L112

What do you thinik? 

Also, does `pathlib` need to be stated as a pypromice dependency now, or is it a standard package with all Python installations?

Reply:
`pathlib` is part of python's standard library.

I was not aware that the "NUMBER" part also needed to be removed. Then I suggest we use regular expressions instead:
```
out_fn = re.sub(r'\d*\.dat$', '', attachment_name) + '.txt'
```

`re` is also part of the standard library.

Reply:
Done! 
https://github.com/GEUS-Glaciology-and-Climate/pypromice/blob/e35459900ca40d9b8771365422e6976e5b1bc93e/src/pypromice/tx/get_watsontx.py#L109-L111